src/protocol/device_path.rs

```
@@ -77,16 +78,18 @@ impl Protocol for DevicePathToTextProtocol {
 }
 
 impl DevicePathToTextProtocol {
-    pub fn device_path_node_to_text(&self, device_node: *const DevicePathProtocol, display_only: bool, allow_shortcuts: bool) -> *const u16 {
-        unsafe {
-            (self.device_path_node_to_text)(device_node, display_only as u8, allow_shortcuts as u8)
-        }
+    pub fn device_path_node_to_text(&self, device_node: *const DevicePathProtocol, display_only: bool, allow_shortcuts: bool) -> Result<&str, ()> {
+        let chars: *const u16 = unsafe { (self.device_path_node_to_text)(device_node, display_only as u8, allow_shortcuts as u8) };
+        let out = utf16_ptr_to_str(chars);
+        ::get_system_table().boot_services().free_pool(chars);
+        out.map_err(|_| { () })
```

Why swallow the underlying error?